packages/at_onboarding_cli/lib/src/onboard/at_onboarding_service_impl.dart

```diff
@@ -339,41 +329,48 @@ class AtOnboardingServiceImpl implements AtOnboardingService {
   ///If not, wait until secondary is created
   Future<void> _waitUntilSecondaryCreated() async {
     final maxRetries = 50;
-    int _retryCount = 0;
-    SecondaryAddress? _secondaryAddress;
+    int retryCount = 1;
+    SecondaryAddress? secondaryAddress;
     SecureSocket? secureSocket;
 
-    while (_retryCount < maxRetries && _secondaryAddress == null) {
-      logger.finer('retrying find secondary.......$_retryCount/$maxRetries');
+    while (retryCount <= maxRetries && secondaryAddress == null) {
+      await Future.delayed(Duration(seconds: 3));
+      logger.finer('retrying find secondary.......$retryCount/$maxRetries');
       try {
-        _secondaryAddress =
-        await _atLookup?.secondaryAddressFinder.findSecondary(_atSign);
+        secondaryAddress =
+            await _atLookup?.secondaryAddressFinder.findSecondary(_atSign);
       } on Exception catch (e) {
         logger.finer(e);
       }
-      _retryCount++;
+      retryCount++;
+    }
+
+    if (secondaryAddress == null){
+      logger.severe('Could not find secondary address for $_atSign');
+      exit(1);
     }
     //resetting retry counter to be used for different operation
-    _retryCount = 0;
-    while (secureSocket == null && _retryCount < maxRetries) {
-      logger.finer('retrying connect secondary.......$_retryCount/$maxRetries');
-      stdout.writeln('Connecting to secondary ...');
+    retryCount = 1;
+    while (secureSocket == null && retryCount <= maxRetries) {
+      await Future.delayed(Duration(seconds: 3));
+      logger.finer('retrying connect secondary.......$retryCount/$maxRetries');
+      stdout.writeln('Connecting to secondary ...$retryCount/$maxRetries');
       try {
         secureSocket = await SecureSocket.connect(
-            _secondaryAddress!.host, _secondaryAddress.port);
+            secondaryAddress.host, secondaryAddress.port,
+            timeout: Duration(seconds: 30)); // 30-second timeout should be enough even for slow networks
```

Comment:
check for secureSocket.remoteAddress and port !=null after SecureSocket.connect. Then set a bool flag. use the bool flag in the while instead of secureSocket == null

Reply:
addressed review comments as part of 3ecbf3191d2a90e5b21cdf9aa8b6da7421e78437